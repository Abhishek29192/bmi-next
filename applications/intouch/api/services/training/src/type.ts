import { DataSources } from "apollo-server-core/dist/graphqlOptions";
import { Docebo } from "./apis";

export type ITokenInfo = {
  token: string;
};

export interface IDataSources extends DataSources<ITokenInfo> {
  doceboApi: Docebo;
}

export interface IContext {
  token: string;
  dataSources?: IDataSources;
}

export interface ISelectOrgchart {
  [branchId: string]: string;
}
export interface ISelectManager {
  [managerId: string]: string;
}
export interface IUserCreateInput {
  userid: string;
  email: string;
  password: string;
  privacy?: string;
  firstname: string;
  lastname: string;
  force_change?: number;
  level?: number;
  language?: string;
  expiration?: string;
  email_validation_status?: number;
  valid?: number;
  date_format?: string;
  timezone?: string;
  role?: number;
  send_notification_email?: boolean;
  can_manage_subordinates?: boolean;
  select_orgchart: ISelectOrgchart;
}
export interface IUserUpdateInput {
  userid: string;
  email: string;
  password: string;
  firstname: string;
  lastname: string;
  force_change?: number;
  level?: number;
  language?: string;
  expiration?: string;
  email_validation_status?: number;
  date_format?: string;
  timezone?: string;
  role?: number;
  manager?: ISelectManager;
  can_manage_subordinates?: boolean;
  select_orgchart: ISelectOrgchart;
  country?: number;
}
export interface IMessage {
  id: string;
  message: string;
}

export interface IUserCreateResponse {
  success: boolean;
  user_id: number;
  message: [IMessage];
}

export interface IUserUpdateResponse {
  success: boolean;
  updated_id: number;
  message: [IMessage];
}

export interface IRulesList {
  condition_value: string;
  condition_type: string;
  id_field_common: number;
}

export interface IGroupCreateInput {
  name: string;
  description: string;
  assign_rules?: number;
  rules_logic?: string;
  assign_existing_users?: boolean;
  rules_list?: [IRulesList];
  id_users?: [number];
}

export interface ITrainingUser {
  id: number;
  username: string;
  firstname: string;
  lastname: string;
  email: string;
  user_level: string;
  userInfo?: any;
  enrollment?: any;
}
export interface ISession {
  data: ITrainingUser;
}
export interface IPageQueryOptions {
  paginate?: IPaginateOptions;
}

export interface IPaginateOptions {
  page?: number;
  page_size?: number;
}
export interface SortOptions {
  field?: string;
  order?: SortOrderEnum;
}

export enum SortOrderEnum {
  Asc = "ASC",
  Desc = "DESC"
}
export type QueryStringParams = {
  [param: string]: string;
};
