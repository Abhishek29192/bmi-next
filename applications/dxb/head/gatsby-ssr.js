import "./src/styles/global.css";
import React from "react";
import BmiThemeProvider from "@bmi/theme-provider";

export const wrapRootElement = ({ element }) => {
  return (
    <BmiThemeProvider includeCssBaseline={false}>{element}</BmiThemeProvider>
  );
};
