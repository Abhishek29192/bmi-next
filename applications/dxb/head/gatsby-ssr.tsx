import { HubspotProvider } from "@aaronhayes/react-use-hubspot-form";
import { ThemeProvider } from "@bmi-digital/components";
import type { GatsbySSR } from "gatsby";
import React from "react";
import { ConfigProvider } from "./src/contexts/ConfigProvider";
import "./src/styles/global.css";

export const wrapRootElement: GatsbySSR["wrapRootElement"] = ({ element }) => {
  return (
    /*providers with static data*/
    <ConfigProvider>
      <ThemeProvider includeCssBaseline={false}>
        <HubspotProvider async={false} addToHead={true}>
          {element}
        </HubspotProvider>
      </ThemeProvider>
    </ConfigProvider>
  );
};
