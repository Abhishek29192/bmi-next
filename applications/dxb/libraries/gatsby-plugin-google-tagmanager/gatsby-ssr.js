"use strict";

exports.onRenderBody = require("./dist/src/gatsby-ssr.js").onRenderBody;
exports.onPreRenderHTML = require("./dist/src/gatsby-ssr.js").onPreRenderHTML;
