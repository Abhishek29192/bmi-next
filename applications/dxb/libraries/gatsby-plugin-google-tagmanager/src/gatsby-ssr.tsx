import { oneLine, stripIndent } from "common-tags";
import React from "react";
import type { GatsbySSR } from "gatsby";
import type { Options } from "./types";

const generateGTMIframe = (
  id: string,
  environmentParamStr: string,
  selfHostedOrigin: string
) =>
  oneLine`<iframe src="${selfHostedOrigin}/ns.html?id=${id}${environmentParamStr}" height="0" width="0" style="display: none; visibility: hidden" aria-hidden="true"></iframe>`;

export const onRenderBody: NonNullable<GatsbySSR["onRenderBody"]> = (
  { setHeadComponents, setPreBodyComponents },
  {
    ids,
    includeInDevelopment = false,
    gtmAuth,
    gtmPreview,
    defaultDataLayer,
    dataLayerName = "dataLayer",
    enableWebVitalsTracking = false,
    selfHostedOrigin = "https://www.googletagmanager.com"
  }: Options
) => {
  if (process.env.NODE_ENV === "production" || includeInDevelopment) {
    const environmentParamStr =
      gtmAuth && gtmPreview
        ? oneLine`
      &gtm_auth=${gtmAuth}&gtm_preview=${gtmPreview}&gtm_cookies_win=x
    `
        : "";

    selfHostedOrigin = selfHostedOrigin.replace(/\/$/, "");

    const inlineScripts: React.ReactNode[] = [];
    const preBodyComponents: React.ReactNode[] = [];

    if (enableWebVitalsTracking) {
      // web-vitals/polyfill (necessary for non chromium browsers)
      // @seehttps://www.npmjs.com/package/web-vitals#how-the-polyfill-works
      inlineScripts.push(
        <script
          key={`gatsby-plugin-google-tagmanager-web-vitals`}
          data-gatsby="web-vitals-polyfill"
          dangerouslySetInnerHTML={{
            __html: `
              !function(){var e,t,n,i,r={passive:!0,capture:!0},a=new Date,o=function(){i=[],t=-1,e=null,f(addEventListener)},c=function(i,r){e||(e=r,t=i,n=new Date,f(removeEventListener),u())},u=function(){if(t>=0&&t<n-a){var r={entryType:"first-input",name:e.type,target:e.target,cancelable:e.cancelable,startTime:e.timeStamp,processingStart:e.timeStamp+t};i.forEach((function(e){e(r)})),i=[]}},s=function(e){if(e.cancelable){var t=(e.timeStamp>1e12?new Date:performance.now())-e.timeStamp;"pointerdown"==e.type?function(e,t){var n=function(){c(e,t),a()},i=function(){a()},a=function(){removeEventListener("pointerup",n,r),removeEventListener("pointercancel",i,r)};addEventListener("pointerup",n,r),addEventListener("pointercancel",i,r)}(t,e):c(t,e)}},f=function(e){["mousedown","keydown","touchstart","pointerdown"].forEach((function(t){return e(t,s,r)}))},p="hidden"===document.visibilityState?0:1/0;addEventListener("visibilitychange",(function e(t){"hidden"===document.visibilityState&&(p=t.timeStamp,removeEventListener("visibilitychange",e,!0))}),!0);o(),self.webVitals={firstInputPolyfill:function(e){i.push(e),u()},resetFirstInputPolyfill:o,get firstHiddenTime(){return p}}}();
            `
          }}
        />
      );
    }

    ids.forEach((id) => {
      inlineScripts.push(
        <script
          key="plugin-google-tagmanager"
          async
          src={`${selfHostedOrigin}/gtm.js?id=${id}${
            dataLayerName !== "dataLayer" ? `&l=${dataLayerName}` : ""
          }${environmentParamStr}`}
        />
      );

      preBodyComponents.push(
        <noscript
          key="plugin-google-tagmanager"
          dangerouslySetInnerHTML={{
            __html: generateGTMIframe(id, environmentParamStr, selfHostedOrigin)
          }}
        />
      );
    });

    inlineScripts.push(
      <script
        key="plugin-google-tagmanager"
        dangerouslySetInnerHTML={{
          __html: stripIndent`
      window.${dataLayerName} = window.${dataLayerName} || [];
      function gtag(){dataLayer.push(arguments);} 
      
      gtag('consent', 'default', { 
        'ad_storage': 'denied', 
        'analytics_storage': 'denied', 
        'functionality_storage': 'denied',
        'personalization_storage': 'denied',
        'security_storage': 'denied',
        'ad_user_data': 'denied',
        'ad_personalization': 'denied',
        'wait_for_update': 500 
      }); 
 
      window.${dataLayerName}.push(${JSON.stringify(defaultDataLayer)});
      gtag('js', new Date());
      [${ids.map((id) => "'" + id + "'")}].forEach((id) => gtag("config", id))
    `
        }}
      />
    );

    setHeadComponents(inlineScripts);
    setPreBodyComponents(preBodyComponents);
  }
};

export const onPreRenderHTML: NonNullable<GatsbySSR["onPreRenderHTML"]> = ({
  getHeadComponents,
  replaceHeadComponents
}) => {
  const headComponents = getHeadComponents() as React.ReactElement[];
  //Needed to make sure that gtag scripts are being loaded at the very beginning
  //In order setup default consent rules
  headComponents.sort((a, b) => {
    if (
      a.key !== "plugin-google-tagmanager" &&
      b.key === "plugin-google-tagmanager"
    ) {
      return 1;
    }

    if (
      a.key === "plugin-google-tagmanager" &&
      b.key !== "plugin-google-tagmanager"
    ) {
      return -1;
    }

    return 0;
  });
  replaceHeadComponents(headComponents);
};
