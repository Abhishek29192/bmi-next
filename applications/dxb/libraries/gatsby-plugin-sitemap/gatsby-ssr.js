"use strict";

var React = _interopRequireWildcard(require("react"));
var _gatsby = require("gatsby");
var _path = require("path");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
// TODO: Remove for v3 - Fix janky path/asset prefixing
var withPrefix = _gatsby.withAssetPrefix || _gatsby.withPrefix;
exports.onRenderBody = function (_ref, pluginOptions) {
  var setHeadComponents = _ref.setHeadComponents;
  var output = pluginOptions.output,
    createLinkInHead = pluginOptions.createLinkInHead;
  if (!createLinkInHead) {
    return;
  }
  setHeadComponents([/*#__PURE__*/React.createElement("link", {
    key: "gatsby-plugin-sitemap",
    rel: "sitemap",
    type: "application/xml",
    href: withPrefix(_path.posix.join(output, "/sitemap-index.xml"))
  })]);
};