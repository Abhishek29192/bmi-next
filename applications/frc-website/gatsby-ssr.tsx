import ThemeProvider from "@bmi-digital/components/theme-provider";
import React from "react";
import type { GatsbySSR } from "gatsby";

export const wrapRootElement: GatsbySSR["wrapRootElement"] = ({ element }) => {
  return <ThemeProvider>{element}</ThemeProvider>;
};
