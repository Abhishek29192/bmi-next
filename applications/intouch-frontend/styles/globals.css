html,
body {
  padding: 0;
  margin: 0;
  -webkit-font-smoothing: antialiased;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

/* TODO: Create proper component for Avatar */
.avatar {
  background-color: #c5c5c5;
  border-radius: 100%;
  width: 70px;
  height: 70px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 0.5rem;
}
