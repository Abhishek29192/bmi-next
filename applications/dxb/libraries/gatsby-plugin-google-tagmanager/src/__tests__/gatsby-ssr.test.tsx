import { describe, it, expect, jest } from "@jest/globals";
import { oneLine } from "common-tags";
import { ReactElement } from "react";
import { onRenderBody } from "../gatsby-ssr";
import type { RenderBodyArgs } from "gatsby";
import type { Options } from "../types";

const createPluginOptions = (pluginOptions?: Partial<Options>): Options => ({
  plugins: [],
  ids: ["123"],
  includeInDevelopment: false,
  routeChangeEventName: "gatsby-route-change",
  enableWebVitalsTracking: false,
  selfHostedOrigin: "https://www.googletagmanager.com",
  ...pluginOptions
});

const createRenderBodyArgs = (
  setHeadComponents: RenderBodyArgs["setHeadComponents"],
  setPreBodyComponents: RenderBodyArgs["setPreBodyComponents"]
): RenderBodyArgs => ({
  setHeadComponents,
  setPreBodyComponents,
  loadPageDataSync: jest.fn<RenderBodyArgs["loadPageDataSync"]>(),
  pathname: "path-name",
  setHtmlAttributes: jest.fn<RenderBodyArgs["setHtmlAttributes"]>(),
  setBodyAttributes: jest.fn<RenderBodyArgs["setBodyAttributes"]>(),
  setPostBodyComponents: jest.fn<RenderBodyArgs["setPostBodyComponents"]>(),
  setBodyProps: jest.fn() // Doesn't like being typed
});

describe("gatsby-plugin-google-tagmanager", () => {
  it("should load gtm", () => {
    const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
    const setPreBodyComponents =
      jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
    const pluginOptions = createPluginOptions({
      includeInDevelopment: true
    });

    onRenderBody(
      createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
      pluginOptions
    );
    const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
    const [preBodyConfig] = setPreBodyComponents.mock
      .calls[0][0] as ReactElement[];

    expect(headConfig!.props.dangerouslySetInnerHTML.__html).toMatchSnapshot();
    expect(
      preBodyConfig!.props.dangerouslySetInnerHTML.__html
    ).toMatchSnapshot();
    // check if no newlines were added
    expect(preBodyConfig!.props.dangerouslySetInnerHTML.__html).not.toContain(
      "\n"
    );
  });

  describe("defaultDatalayer", () => {
    it("should add no dataLayer by default", () => {
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );
      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      // eslint-disable-next-line no-useless-escape
      expect(headConfig!.props.dangerouslySetInnerHTML.__html).not.toContain(
        "window.dataLayer"
      );
      expect(headConfig!.props.dangerouslySetInnerHTML.__html).not.toContain(
        "undefined"
      );
    });

    it("should add a static object as defaultDatalayer", () => {
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true,
        defaultDataLayer: {
          type: "object",
          value: { pageCategory: "home" }
        }
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );
      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      expect(
        headConfig!.props.dangerouslySetInnerHTML.__html
      ).toMatchSnapshot();
      expect(headConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        "window.dataLayer"
      );
    });

    it("should add a function as defaultDatalayer", () => {
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true,
        defaultDataLayer: {
          type: "function",
          value: function () {
            return { pageCategory: window.pageType };
          }.toString()
        }
      });

      const datalayerFuncAsString = oneLine`${
        pluginOptions.defaultDataLayer!.value
      }`;

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );
      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      expect(
        headConfig!.props.dangerouslySetInnerHTML.__html
      ).toMatchSnapshot();
      expect(headConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        `window.dataLayer.push((${datalayerFuncAsString})());`
      );
    });

    it("should add a renamed dataLayer", () => {
      const dataLayerName = "TEST_DATA_LAYER_NAME";
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true,
        defaultDataLayer: {
          type: "object",
          value: { pageCategory: "home" }
        },
        dataLayerName,
        enableWebVitalsTracking: false
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );
      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      expect(
        headConfig!.props.dangerouslySetInnerHTML.__html
      ).toMatchSnapshot();
      expect(headConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        `window.${dataLayerName}`
      );
    });

    it("adds the web-vitals polyfill to the head", () => {
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true,
        enableWebVitalsTracking: true
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );

      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      expect(setHeadComponents.mock.calls[0][0].length).toBe(2);
      expect(headConfig!.key).toBe(
        "gatsby-plugin-google-tagmanager-web-vitals"
      );
    });

    it("should not add the web-vitals polyfill when enableWebVitalsTracking is false", () => {
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true,
        enableWebVitalsTracking: false
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );

      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      expect(setHeadComponents.mock.calls[0].length).toBe(1);
      expect(headConfig!.key).toBe("plugin-google-tagmanager");
    });

    it("should set selfHostedOrigin as googletagmanager.com by default", () => {
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );
      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      const [preBodyConfig] = setPreBodyComponents.mock
        .calls[0][0] as ReactElement[];

      // eslint-disable-next-line no-useless-escape
      expect(headConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        "https://www.googletagmanager.com/gtm.js"
      );
      expect(preBodyConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        "https://www.googletagmanager.com/ns.html"
      );
    });

    it("should set selfHostedOrigin", () => {
      const selfHostedOrigin = "YOUR_SELF_HOSTED_ORIGIN";
      const setHeadComponents = jest.fn<RenderBodyArgs["setHeadComponents"]>();
      const setPreBodyComponents =
        jest.fn<RenderBodyArgs["setPreBodyComponents"]>();
      const pluginOptions = createPluginOptions({
        includeInDevelopment: true,
        selfHostedOrigin: selfHostedOrigin
      });

      onRenderBody(
        createRenderBodyArgs(setHeadComponents, setPreBodyComponents),
        pluginOptions
      );
      const [headConfig] = setHeadComponents.mock.calls[0][0] as ReactElement[];
      const [preBodyConfig] = setPreBodyComponents.mock
        .calls[0][0] as ReactElement[];

      expect(headConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        `${selfHostedOrigin}/gtm.js`
      );
      expect(preBodyConfig!.props.dangerouslySetInnerHTML.__html).toContain(
        `${selfHostedOrigin}/ns.html`
      );
    });
  });
});
