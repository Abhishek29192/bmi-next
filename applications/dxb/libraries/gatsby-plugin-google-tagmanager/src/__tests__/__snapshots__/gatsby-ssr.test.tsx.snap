// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-google-tagmanager should load gtm 1`] = `
"window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);} 

gtag('consent', 'default', { 
  'ad_storage': 'denied', 
  'analytics_storage': 'denied', 
  'functionality_storage': 'denied',
  'personalization_storage': 'denied',
  'security_storage': 'denied',
  'ad_user_data': 'denied',
  'ad_personalization': 'denied',
  'wait_for_update': 500 
}); 
 
window.dataLayer.push({"platform":"gatsby","env":"production"});
gtag('js', new Date().getTime());
['123'].forEach((id) => gtag("config", id))"
`;

exports[`gatsby-plugin-google-tagmanager should load gtm 2`] = `"<iframe src="https://www.googletagmanager.com/ns.html?id=123" height="0" width="0" style="display: none; visibility: hidden" aria-hidden="true"></iframe>"`;
